src/pybind/mgr/stats/fs/perf_stats.py

```diff
@@ -166,6 +172,47 @@ def notify(self, cmdtag):
             self.log.debug("client_metadata={0}, to_purge={1}".format(
                 self.client_metadata['metadata'], self.client_metadata['to_purge']))
 
+    def notify_fsmap(self):
+        #Reregister the user queries when there is a new rank0 mds
+        gid_state = FSPerfStats.get_rank0_mds_gid_state(self.module.get('fs_map'))
+        if not gid_state:
+            return
+        rank0_gid, state = gid_state
+        if (rank0_gid and (rank0_gid != self.prev_rank0_gid)):
+            if (state != 'up:active'):
+                return
+            #ronk0 MDS is up:active
+            ua_last_updated = datetime.now()
+            Timer(0, self.re_register_queries, args=(rank0_gid, ua_last_updated,)).start()
+
+    def re_register_queries(self, rank0_gid, ua_last_updated):
+        while True:
```

Comment:
This keeps the timer task pinned to this job - if there were other timer tasks queues, they would never get executed until this finishes. This should reschedule itself when it cannot proceed. Also, condition variables are to be used with a mutex.